src/sysv/systemd/sssd-kcm.service.in

```diff
@@ -12,8 +12,8 @@ Environment=DEBUG_LOGGER=--logger=files
 ExecStartPre=+-/bin/chown -f @SSSD_USER@:@SSSD_USER@ @sssdconfdir@
 ExecStartPre=+-/bin/chown -f @SSSD_USER@:@SSSD_USER@ @sssdconfdir@/sssd.conf
 ExecStartPre=+-/bin/chown -f -R @SSSD_USER@:@SSSD_USER@ @sssdconfdir@/conf.d
-ExecStartPre=+-/bin/chown -f -R @SSSD_USER@:@SSSD_USER@ @secdbpath@/*.ldb
-ExecStartPre=+-/bin/chown -f -R @SSSD_USER@:@SSSD_USER@ @logpath@/sssd_kcm.log
+ExecStartPre=+-/bin/sh -c "/bin/chown -f @SSSD_USER@:@SSSD_USER@ @secdbpath@/*.ldb"
```

Comment:
Hi,

instead of spawning a shell would something like

   find @secdbpath@ -type -f -iname '*.ldb' -exec  /bin/chown -f @SSSD_USER@:@SSSD_USER@ {} \;

work (I have not tested this).

Additionally, what about the `*ccache_*` files in this directory. Since the child processes accessing them have higher privileges to read the keytab the actual permissions might currently not be important. But it might be good to have an understanding what should be the expected permissions.

bye,
Sumit

Reply:
Hmm interesting,

according the documentation +- should do the trick. Looks like systemd issue (either code or documentation).

I like the `find` solution. I often use this this when I'm not sure that files are present.

Tom

Reply:
> according the documentation +- should do the trick. Looks like systemd issue (either code or documentation).

Systemd simply execs 'chown' providing '*.ldb' "as is".
'chown' doesn't handle '*', shell does.

Reply:
> instead of spawning a shell would something like
> 
> find @secdbpath@ -type -f -iname '*.ldb' -exec /bin/chown -f @SSSD_USER@:@SSSD_USER@ {} ;

What's wrong with shell?
I'm afraid systemd will mangle something in so complicated string.

Reply:
Ahh, got it. "+-" works, just the star (*) does not. Lets use the find solution

Reply:
> What's wrong with shell?

I mean, 'find -exec' forks anyways.
What are drawbacks of shell?

Reply:
> Additionally, what about the `*ccache_*` files in this directory. Since the child processes accessing them have higher privileges to read the keytab the actual permissions might currently not be important. But it might be good to have an understanding what should be the expected permissions.

`@SSSD_USER@:@SSSD_USER@` as well?

Reply:
the difference is that if there is no file, the * is not expanded and `rm` tries to delete file named `*.log` which is not present. Hence it produces ugly message sometimes and fails. But both approaches works.

Reply:
'-f' should suppress error messages and '-' ignores exit code.

Reply:
> > Additionally, what about the `*ccache_*` files in this directory. Since the child processes accessing them have higher privileges to read the keytab the actual permissions might currently not be important. But it might be good to have an understanding what should be the expected permissions.
> 
> `@SSSD_USER@:@SSSD_USER@` as well?

On the other hand, all this clutter is to facilitate one-time-event  --  upgrade to sssd-2.10 / change of default service user.
By the time ldap_/krb5_ child lose CAP_DAC_OVERRIDE (if ever) it should be already possible to remove all those chown's from service descriptions...
So frankly I don't think it's worth it.

Reply:
Hi,
yes, my question was just about the expected permissions, no need to change those at startup.

bye,
Sumit

Reply:
I'm fine either way. If we use find, we have to require findutils (we currently require it only for build time), which is probably fine, but still a dependency.

Reply:
Solution with 'find' - #7598

Both works:
```
    Process: 5769 ExecStartPre=/bin/sh -c /bin/chown -f sssd:sssd /var/lib/sss/db/*.ldb (code=exited, status=0/SUCCESS)
    Process: 5771 ExecStartPre=/bin/sh -c /bin/chown -f sssd:sssd /var/log/sssd/*.log (code=exited, status=0/SUCCESS)
```
and
```
    Process: 6245 ExecStartPre=find /var/lib/sss/db/ -type f -iname *.ldb -exec /bin/chown -f sssd:sssd {} ; (code=exited, status=0/SUCCESS)
    Process: 6251 ExecStartPre=find /var/log/sssd/ -type f -iname *.log -exec /bin/chown -f sssd:sssd {} ; (code=exited, status=0/SUCCESS)
```

I personally prefer 'sh' variant as it's easier to read (for me) but I don't strongly oppose 'find' variant either.

Reply:
Hi,

I suggested the find variant because calling a shell might have some unknown side-effects. But it looks like other services are using shells as well in `ExecStart*` . So feel free to use the shell variant.

bye,
Sumit